python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -1458,9 +1458,21 @@ def open_server_process(
         subprocess_args += loadable_target_origin.get_cli_args()
 
     env = {
-        **(env or os.environ),
+        **os.environ,
     }
 
+    if (
+        executable_path
+        and is_likely_venv_executable(executable_path)
+        and executable_path != sys.executable
+    ):
+        # ensure that if a venv is being used as an executable path that the same PATH
+        # that would be set if the venv was activated is also set in the launched
+        # subprocess
+        current_path = os.environ.get("PATH")
+        added_path = os.path.dirname(executable_path)
+        env["PATH"] = f"{added_path}{os.pathsep}{current_path}" if current_path else added_path
```

Comment:
some thoughts 

should we check if its already here before adding it again 
* feels like avoiding PATH manipulation when possible is nice 
* it being duplicated doesn't shouldn't change behavior 

should we add it to the back instead of the front 
* pro: avoids breaking anybodies current PATH resolution 
* con: is different than normal venv activation

Reply:
i think we should match what it would be if you activated it before running, yeah. which kind of addresses the 'should we add it if it's already there question' too - we should strive to match behavior and the order could matter.